src/server/checker/CheckerService.ts

```diff
@@ -68,19 +79,79 @@ export class CheckerService {
     id,
     user
   ) => {
-    const result = await this.CheckerModel.findByPk(
-      id,
-      user ? { include: [this.UserModel] } : undefined
-    )
-    if (result && user) {
-      const isAuthorized = result.users?.some((u) => u.id === user.id)
-      if (!isAuthorized) {
-        const resultWithoutUsers: Checker = { ...(result.toJSON() as Checker) }
-        delete resultWithoutUsers.users
-        return resultWithoutUsers
+    return await this.findAndCheckAuth(id, user)
+  }
+
+  retrievePublished: (
+    id: string
+  ) => Promise<GetPublishedCheckerWithoutDraftCheckerDTO | undefined> = async (
+    id
+  ) => {
+    const result = await this.PublishedCheckerModel.findOne({
+      include: [
+        {
+          model: this.CheckerModel,
+          required: true, // Required for an inner join; otherwise the query returns a left outer join
+          where: { id },
+        },
+      ],
+      order: [['createdAt', 'DESC']],
+    })
+
+    // If a publishedChecker is found, we want to remove the draft checker from the publishedChecker object
+    // since anyone could retrieve it without auth
+    if (result) {
+      const publishedCheckerWithoutDraftChecker: GetPublishedCheckerWithoutDraftCheckerDTO = {
+        ...(result.toJSON() as Checker),
       }
+      publishedCheckerWithoutDraftChecker.id =
+        publishedCheckerWithoutDraftChecker.checkerId
+      delete publishedCheckerWithoutDraftChecker.checker
+      delete publishedCheckerWithoutDraftChecker.checkerId
+      return publishedCheckerWithoutDraftChecker as GetPublishedCheckerWithoutDraftCheckerDTO
+    }
+  }
+
+  publish: (
+    id: string,
+    checker: Checker,
+    user: User
+  ) => Promise<Checker> = async (id, checker, user) => {
+    const transaction = await this.sequelize.transaction()
+    const transactionOptions = this.isSqliteFile ? {} : { transaction }
+
+    try {
+      const existingChecker = await this.findAndCheckAuth(id, user)
+      if (!existingChecker) throw new Error(`Checker ${id} not found`)
+
+      // Perform update in checkers and create in publishedCheckers with 1 transaction
+      await this.CheckerModel.update(checker, {
+        where: { id },
+        ...transactionOptions,
+      })
+
+      const createPublishedChecker: CreatePublishedCheckerDTO = {
+        title: checker.title,
+        description: checker.description,
+        fields: checker.fields,
+        constants: checker.constants,
+        operations: checker.operations,
+        displays: checker.displays,
+        checkerId: id, // we want to make sure that we are publishing the right checker
+      }
+
+      await this.PublishedCheckerModel.create(createPublishedChecker, {
+        include: [this.CheckerModel],
```

Comment:
```suggestion
```
From what I get from the [docs](https://sequelize.org/master/manual/creating-with-associations.html), I think this might not be needed since we're not creating the Checker instance also with this call.

Reply:
Fixed in 0882214